### PR TITLE
Create api server

### PR DESCRIPTION
**Pull request by liamchampton**
In this PR
- Dockerfile
- RESTful API routes for creating a person in the DB, getting all people in the DB and an extracted delete person from DB route
- Documentation on the routes

### CODE REVIEW

File: fundamentals-section/server/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Sever setup
```
Comment (adelina-simion-form3) [inline fundamentals-section/server/README.md:1]:
```suggestion
# Server setup
```

File: fundamentals-section/server/main.go [hunk 2]
```diff
@@ -1,20 +1,180 @@
...
+		switch r.Method {
+		case "GET":
+			getAllPeople(w, r, client)
+		case "POST":
+			createPerson(w, r, client)
+		case "OPTIONS":
+			// Set necessary headers for handling CORS requests
+			w.Header().Set("Access-Control-Allow-Origin", "*")
+			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
+			w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
```
Comment (adelina-simion-form3) [inline fundamentals-section/server/main.go:61]: I needs these lines on GET and POST as well please

File: fundamentals-section/server/main.go [hunk 3]
```diff
@@ -1,20 +1,180 @@
...
\ No newline at end of file
+	// Insert the new person document into the collection
+	_, err = collection.InsertOne(context.Background(), &person)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Set the response status code to 201 Created and return the newly created person's ID
+	w.WriteHeader(http.StatusCreated)
```
Comment (adelina-simion-form3) [inline fundamentals-section/server/main.go:107]: Missing content type here `w.Header().Set("Content-Type", "application/json; charset=UTF-8")`

File: fundamentals-section/server/main.go [hunk 4]
```diff
@@ -1,20 +1,180 @@
...
+		err = cursor.Decode(&person)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		people = append(people, person)
+	}
+
+	// Return the people slice as JSON
+	json.NewEncoder(w).Encode(people)
```
Comment (adelina-simion-form3) [inline fundamentals-section/server/main.go:179]: Missing content type here `w.Header().Set("Content-Type", "application/json; charset=UTF-8")`

**Review by adelina-simion-form3 — CHANGES_REQUESTED** [review]

File: fundamentals-section/server/main.go (hunk 2 above)
Comment (liamchampton) [reply fundamentals-section/server/main.go:61]: I have moved them outside of the switch statement so the code is not duplicated 3x

File: fundamentals-section/server/main.go (hunk 4 above)
Comment (liamchampton) [reply fundamentals-section/server/main.go:179]: Returns:
```json
curl -i -X GET localhost:8080/persons
HTTP/1.1 200 OK
....
Date: Mon, 13 Mar 2023 09:58:45 GMT
Content-Length: 1387
Content-Type: text/plain; charset=utf-8

[{"id":"6408c175091d45f5fa24f0f6","name":"Liam Hampton","fac....
```

File: fundamentals-section/server/main.go (hunk 3 above)
Comment (liamchampton) [reply fundamentals-section/server/main.go:107]: Returns:
```json
curl -i -X POST -H "Content-Type: application/json" -d '{
    "name": "Jenny Twirl",
    "fact": "Likes to twirl",
    "location": "Egypt"
}' localhost:8080/persons
HTTP/1.1 201 Created
....
Content-Type: application/json; charset=UTF-8
Date: Mon, 13 Mar 2023 10:02:04 GMT
Content-Length: 34

{"ID":"640...."}
```

**Review by adelina-simion-form3 — APPROVED** [review]

**Pull request merged**